core/utils/fork-tree/src/lib.rs

```diff
@@ -75,7 +75,7 @@ pub enum FinalizationResult<V> {
 /// in order. Each node is uniquely identified by its hash but can be ordered by
 /// its number. In order to build the tree an external function must be provided
 /// when interacting with the tree to establish a node's ancestry.
-#[derive(Clone, Debug, Decode, Encode, PartialEq)]
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Default)]
 pub struct ForkTree<H, N, V> {
```

Comment:
do you use the default ? I'm not sure how interesting a default is here.

Reply:
and I wonder what is even the rational of clippy recommending it? just because you can?

Reply:
I think the rationale is because this type provides `fn new() -> Self` then according to https://rust-lang.github.io/api-guidelines/predictability.html#constructors-are-static-inherent-methods-c-ctor it is convient to provide `default` as well,